index.js

```diff
@@ -4,65 +4,47 @@ import domMutations from 'dom-mutations';
 const isDomReady = target =>
 	['interactive', 'complete'].includes((target.ownerDocument ?? target).readyState);
 
-export default function elementReady(selector, {
+export default async function elementReady(selector, {
 	target = document,
 	stopOnDomReady = true,
 	waitForChildren = true,
-	timeout = Number.POSITIVE_INFINITY,
+	signal,
 	predicate,
 } = {}) {
 	// Not necessary, it just acts faster and avoids listener setup
```

Comment:
I would add a `if (signal?.aborted) {
			return;
		}` check here too, to finish as fast as posisble when signal is already aborted.